module/zfs/fm.c

```diff
@@ -593,8 +593,9 @@ zfs_zevent_fd_hold(int fd, minor_t *minorp, zfs_zevent_t **ze)
 	if (fp == NULL)
 		return (EBADF);
 
-	*minorp = zfsdev_getminor(fp->f_file);
-	error = zfs_zevent_minor_to_state(*minorp, ze);
+	error = zfsdev_getminor(fp->f_file, minorp);
+	if (error == 0)
```

Comment:
Since the line comments disappeared as I pushed a revised pull request, I want to copy what I said in response to `[style] The above two if's should just be an if-else block.`:

```
If we do that, then an error in `zfs_zevent_minor_to_state()` would be ignored.
```